File: output/glyphs/candid/prepare.py
from PIL import Image, ImageFilter
import numpy as np, subprocess
from pathlib import Path
out=Path('output/glyphs/candid')
src=Image.open('output/glyphs/candid/source.png').convert('RGB')
a=np.asarray(src);lum=np.asarray(src.convert('L'))
# White foreground is separated from the mid-gray baked checkerboard.
mask=Image.fromarray(np.where(lum>230,255,0).astype('uint8')).filter(ImageFilter.GaussianBlur(.6))
rgba=src.convert('RGBA');rgba.putalpha(mask);rgba.save(out/'candid-transparent.png')
box=mask.getbbox();mark=rgba.crop(box);mark.thumbnail((240,240),Image.Resampling.LANCZOS)
canvas=Image.new('RGBA',(360,360));canvas.alpha_composite(mark,((360-mark.width)//2,(360-mark.height)//2))
frames=[];raw=[]
for i in range(72):
 im=canvas.rotate(-360*i/72,Image.Resampling.BICUBIC,expand=False)
 # Quantize alpha coverage after downsizing; black is empty in the site sampler.
 small=im.resize((88,88),Image.Resampling.LANCZOS)
 alpha=np.asarray(small.getchannel('A'),dtype=float)
 field=(np.rint(alpha/255*5)*51).astype('uint8');raw.append(field.tobytes())
 preview=im.resize((256,256),Image.Resampling.LANCZOS)
 ix=np.where(np.asarray(preview.getchannel('A'))>=128,1,0).astype('uint8')
 pal=Image.fromarray(ix,'P');pal.putpalette([0,0,0,255,255,255]+[0]*762);pal.info['transparency']=0;frames.append(pal)
frames[0].save(out/'candid-rotation.gif',save_all=True,append_images=frames[1:],duration=[40,40,40,50]*18,loop=0,disposal=2,transparency=0,optimize=False)
subprocess.run(['ffmpeg','-v','error','-f','rawvideo','-pix_fmt','gray','-s','88x88','-r','24','-i','pipe:0','-c:v','libvpx-vp9','-lossless','1','-pix_fmt','yuv420p','-y','public/cursors/levels/candid-rotation-L6.webm'],input=b''.join(raw),check=True)
print('72 frames, 3-second full rotation; transparent PNG and GIF saved')

File: output/glyphs/tinaj/prepare.mjs
import {execFileSync as run} from 'node:child_process';
import fs from 'node:fs';
const dir='output/glyphs/tinaj';
for(const n of [240,88]){
 const raw=run('ffmpeg',['-v','error','-i',`${dir}/source.gif`,'-vf',`format=rgba,colorkey=0xFFFFFF:0.08:0,pad=280:280:20:20:color=black@0,scale=${n}:${n}:flags=area`,'-f','rawvideo','-pix_fmt','rgba','-'],{maxBuffer:1<<28});
 let lo=255,hi=0;
 for(let p=0;p<raw.length;p+=4)if(raw[p+3]>=128){let y=.299*raw[p]+.587*raw[p+1]+.114*raw[p+2];lo=Math.min(lo,y);hi=Math.max(hi,y)}
 const gray=Buffer.alloc(raw.length/4);
 for(let p=0,j=0;p<raw.length;p+=4,j++){
 const solid=raw[p+3]>=128;
 const y=.299*raw[p]+.587*raw[p+1]+.114*raw[p+2];
 const k=solid?1+Math.round(Math.max(0,Math.min(1,(y-lo)/(hi-lo)))*4):0;
 gray[j]=k*51;raw[p]=raw[p+1]=raw[p+2]=k*51;raw[p+3]=solid?255:0;
 }
 if(n===88)run('ffmpeg',['-v','error','-f','rawvideo','-pix_fmt','gray','-s','88x88','-r','40','-i','pipe:0','-c:v','libvpx-vp9','-lossless','1','-pix_fmt','yuv420p','-y','public/cursors/levels/tinaj-gimbal-L6.webm'],{input:gray});
 else run('ffmpeg',['-v','error','-f','rawvideo','-pix_fmt','rgba','-s','240x240','-r','40','-i','pipe:0','-filter_complex','split[a][b];[a]palettegen=max_colors=6:reserve_transparent=1[p];[b][p]paletteuse=dither=none','-loop','0','-y',`${dir}/tinaj-glyph.gif`],{input:raw});
 console.log(n,'pixels;',raw.length/(n*n*4),'frames; grayscale:',[...new Set(gray)].sort((a,b)=>a-b));
}
